src/java/org/apache/cassandra/repair/autorepair/RepairTokenRangeSplitter.java

```diff
@@ -691,41 +688,28 @@ static SizeEstimate getSizesForRangeOfSSTables(AutoRepairConfig.RepairType repai
                 if (metadata != null)
                     cardinality = cardinality.merge(metadata.cardinalityEstimator);
 
-                long sstableSize = reader.bytesOnDisk();
+                // use onDiskLength, which is the actual size of the SSTable data file.
+                long sstableSize = reader.onDiskLength();
```

Comment:
Using `onDiskLength()` which looks at the data file specifically, where bytesOnDisk sums all the components which would cause us to under estimate a little bit potentially.  Ultimately i think either value is probably with in the margins.